src/win32/findfile.c

```diff
@@ -67,6 +67,48 @@ static wchar_t* win32_walkpath(wchar_t *path, wchar_t *buf, size_t buflen)
 	return (path != base) ? path : NULL;
 }
 
+static int win32_is_git_root(wchar_t *root)
+{
+	/* decide whether this is http://git-for-windows.github.io or http://msysgit.github.io
+	* in git-for-windows the root is either C:\Program Files\Git\mingw64 or C:\Program Files\Git\mingw32, whereas C:\PROGRAMFILES\Git\{cmd,mingw64\bin,mingw32\bin} is on the path
+	* in msysgit the root is C:\Program Files (x86)\Git (on x86 and x64) whereas C:\Program Files (x86)\Git\{bin,cmd} is on the path
```

Comment:
@YueLinHo I'll comment on the images although I believe they should not be part of this discussion.

All except the middle image look fine to me. The middle image seems to imply that you can run Cygwin binaries right away on Linux. That's not the case. You can run (many of) them with Wine on Linux, but that's also true for native Windows programs. In other words, if you intend to run a Windows binary on Linux, there's no benefit of having that binary linked against Cygwin / MSYS or not.